app/controllers/api/v2/base_controller.rb

```diff
@@ -64,6 +65,10 @@ def metadata_total
         @total ||= resource_scope.try(:count).to_i
       end
 
+      def can_create
+        User.current.can?("create_#{controller_name.singularize}")
```

Comment:
Can you please use `controller_permission` here instead of `controller_name.singularize`? We have a helper that can be overridden in controllers (e.g. in plugins that use scoped controllers). Check out the find common concern for where this is defined.

We should also check that the model includes `Authorizable`. Some models might not have authentication enabled and this could break them.